express-notes-api/index.js

```diff
@@ -0,0 +1,127 @@
+const express = require('express');
+
+const app = express();
+
+const data = require('./data.json');
+
+const fs = require('fs');
+
+app.get('/api/notes', (req, res) => {
+  const arrayObjects = [];
+  const notes = data.notes;
+  for (const key in notes) {
+    arrayObjects.push(notes[key]);
+  }
+  res.send(arrayObjects);
+});
+
+app.listen(3000, () => {
+  // eslint-disable-next-line no-console
+  console.log('Express server listening on port 3000');
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const notes = data.notes;
+  const number = Number(req.params.id);
+  if (number <= 0 || !Number.isInteger(number) || number === 'NaN') {
+    res.status(400).send({
+      error: 'id must be a postiive integer'
+    });
+  }
+  if (notes[number] === undefined) {
+    res.status(404).json({
+      error: 'no note with the specified id'
+    });
+  }
+  if (notes[number] !== undefined) {
+    res.status(200).json(notes[number]);
+  }
+});
+
+app.use(express.json());
+
+app.post('/api/notes', (req, res) => {
+  const postObject = req.body.content;
+  const notes = data.notes;
+  if (!postObject) {
+    res.status(400).json({
+      error: 'content is a required field'
+    });
+  }
+  if (postObject) {
+    const test = req.body;
+    test.Id = data.nextId;
+    notes[data.nextId] = Object;
```

Comment:
`Object` is a built-in constructor function. Please make sure you're assigning the actual note data.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object